CMakeLists.txt

```diff
@@ -3,16 +3,47 @@ cmake_minimum_required(VERSION 3.9)
 project(Ginkgo LANGUAGES C CXX VERSION 1.1.1 DESCRIPTION "A numerical linear algebra library targeting many-core architectures")
 set(Ginkgo_VERSION_TAG "develop")
 set(PROJECT_VERSION_TAG ${Ginkgo_VERSION_TAG})
+include(CheckLanguage)
+
+# Determine which languages can be compiled
+check_language(CUDA)
+if(CMAKE_CUDA_COMPILER)
+    set(GINKGO_HAS_CUDA ON)
+else()
+    set(GINKGO_HAS_CUDA OFF)
+endif()
+
+find_package(OpenMP)
+if(OpenMP_CXX_FOUND)
+    set(GINKGO_HAS_OMP ON)
+else()
+    set(GINKGO_HAS_OMP OFF)
+endif()
+
+if(NOT DEFINED HIP_PATH)
+    if(NOT DEFINED ENV{HIP_PATH})
+        set(HIP_PATH "/opt/rocm/hip" CACHE PATH "Path to which HIP has been installed")
+        set(ENV{HIP_PATH} ${HIP_PATH})
+    else()
+        set(HIP_PATH $ENV{HIP_PATH} CACHE PATH "Path to which HIP has been installed")
+    endif()
+endif()
+find_program(GINKGO_HIPCONFIG_PATH hipconfig HINTS "${HIP_PATH}/bin")
+if(GINKGO_HIPCONFIG_PATH)
+    set(GINKGO_HAS_HIP ON)
+else()
+    set(GINKGO_HAS_HIP OFF)
+endif()
```

Comment:
Is it possible to put this block into a separate file and just include it here ?  I think it kind of pollutes the main `CMakeLists.txt` file.

Reply:
Good point, I moved them out. Didn't know that `include()` is macro-like as it doesn't create its own scope, so that works fine.